### ISSUE TITLE
test failed in CI: `integration_tests::metrics::test_mgs_metrics`

### ISSUE THREAD

**Issue by bnaecker** [oxidecomputer/omicron#7084]
This test failed on a CI run on #6966:

https://github.com/oxidecomputer/omicron/pull/6966/checks?check_run_id=33061656597

Log showing the specific test failure:

https://buildomat.eng.oxide.computer/wg/0/details/01JCRN7QYMD5JN7HE3C0X1YXBZ/s4SpNBaZFjeEWyFLHsqSy6SfQjObPNywetWcp1XiylxmK3oM/01JCRN8DYKHNYZJVDTMKWKXM6A#S6445

Excerpt from the log showing the failure:

```
2024-11-15T20:11:34.959Z	--- STDOUT:              omicron-nexus::test_all integration_tests::metrics::test_mgs_metrics ---
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	running 1 test
2024-11-15T20:11:34.959Z	test integration_tests::metrics::test_mgs_metrics ... FAILED
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	failures:
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	failures:
2024-11-15T20:11:34.959Z	    integration_tests::metrics::test_mgs_metrics
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 397 filtered out; finished in 15.48s
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	
2024-11-15T20:11:34.959Z	--- STDERR:              omicron-nexus::test_all integration_tests::metrics::test_mgs_metrics ---
2024-11-15T20:11:34.959Z	log file: /var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.0.log
2024-11-15T20:11:34.959Z	note: configured to log to "/var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.0.log"
2024-11-15T20:11:34.959Z	DB URL: postgresql://root@[::1]:46169/omicron?sslmode=disable
2024-11-15T20:11:34.959Z	DB address: [::1]:46169
2024-11-15T20:11:34.959Z	log file: /var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.2.log
2024-11-15T20:11:34.959Z	note: configured to log to "/var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.2.log"
2024-11-15T20:11:34.959Z	log file: /var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.3.log
2024-11-15T20:11:34.960Z	note: configured to log to "/var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.3.log"
2024-11-15T20:11:34.960Z	log file: /var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.4.log
2024-11-15T20:11:34.960Z	note: configured to log to "/var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.4.log"
2024-11-15T20:11:34.960Z	
2024-11-15T20:11:34.960Z	=== checking timeseries for hardware_component:temperature ===
2024-11-15T20:11:34.960Z	
2024-11-15T20:11:34.960Z	thread 'integration_tests::metrics::test_mgs_metrics' panicked at nexus/tests/integration_tests/metrics.rs:315:13:
2024-11-15T20:11:34.960Z	could not parse timeseries query response: parsing response body
2024-11-15T20:11:34.960Z	
2024-11-15T20:11:34.960Z	Caused by:
2024-11-15T20:11:34.960Z	    missing field `tables` at line 5 column 1
2024-11-15T20:11:34.960Z	
2024-11-15T20:11:34.960Z	Stack backtrace:
2024-11-15T20:11:34.960Z	   0: <E as anyhow::context::ext::StdError>::ext_context
2024-11-15T20:11:34.960Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.93/src/backtrace.rs:27:14
2024-11-15T20:11:34.960Z	   1: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::context
2024-11-15T20:11:34.960Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.93/src/context.rs:54:31
2024-11-15T20:11:34.960Z	   2: nexus_test_utils::http_testing::TestResponse::parsed_body
2024-11-15T20:11:34.964Z	             at ./test-utils/src/http_testing.rs:513:9
2024-11-15T20:11:34.964Z	   3: {async_fn#0}<&&str>
2024-11-15T20:11:34.964Z	             at ./tests/integration_tests/metrics.rs:313:5
2024-11-15T20:11:34.964Z	   4: {async_fn#0}
2024-11-15T20:11:34.964Z	             at ./tests/integration_tests/metrics.rs:693:18
2024-11-15T20:11:34.964Z	   5: {async_block#0}
2024-11-15T20:11:34.964Z	             at ./tests/integration_tests/metrics.rs:659:79
2024-11-15T20:11:34.965Z	   6: {async_fn#0}<(), (), test_all::integration_tests::metrics::test_mgs_metrics::{async_block#0}::test_mgs_metrics::{async_fn#0}::check_all_timeseries_present::{async_fn#0}::{closure_env#1}, test_all::integration_tests::metrics::test_mgs_metrics::{async_block#0}::test_mgs_metrics::{async_fn#0}::check_all_timeseries_present::{async_fn#0}::{closure#1}::{async_block_env#0}>
2024-11-15T20:11:34.965Z	             at /work/oxidecomputer/omicron/test-utils/src/dev/poll.rs:80:28
2024-11-15T20:11:34.965Z	   7: {async_fn#0}
2024-11-15T20:11:34.965Z	             at ./tests/integration_tests/metrics.rs:670:10
2024-11-15T20:11:34.965Z	   8: {async_fn#0}
2024-11-15T20:11:34.965Z	             at ./tests/integration_tests/metrics.rs:744:75
2024-11-15T20:11:34.965Z	   9: {async_block#0}
2024-11-15T20:11:34.965Z	             at ./tests/integration_tests/metrics.rs:530:1
2024-11-15T20:11:34.965Z	  10: poll<&mut dyn core::future::future::Future<Output=()>>
2024-11-15T20:11:34.965Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
2024-11-15T20:11:34.965Z	  11: poll<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.965Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
2024-11-15T20:11:34.965Z	  12: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:57
2024-11-15T20:11:34.965Z	  13: with_budget<core::task::poll::Poll<()>, tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure#0}::{closure#0}::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>>
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:107:5
2024-11-15T20:11:34.965Z	  14: budget<core::task::poll::Poll<()>, tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure#0}::{closure#0}::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>>
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:73:5
2024-11-15T20:11:34.965Z	  15: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:25
2024-11-15T20:11:34.965Z	  16: tokio::runtime::scheduler::current_thread::Context::enter
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:423:19
2024-11-15T20:11:34.965Z	  17: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:695:36
2024-11-15T20:11:34.965Z	  18: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:68
2024-11-15T20:11:34.965Z	  19: tokio::runtime::context::scoped::Scoped<T>::set
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/scoped.rs:40:9
2024-11-15T20:11:34.965Z	  20: tokio::runtime::context::set_scheduler::{{closure}}
2024-11-15T20:11:34.965Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:26
2024-11-15T20:11:34.965Z	  21: try_with<tokio::runtime::context::Context, tokio::runtime::context::set_scheduler::{closure_env#0}<(alloc::boxed::Box<tokio::runtime::scheduler::current_thread::Core, alloc::alloc::Global>, core::option::Option<()>), tokio::runtime::scheduler::current_thread::{impl#8}::enter::{closure_env#0}<tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>, core::option::Option<()>>>, (alloc::boxed::Box<tokio::runtime::scheduler::current_thread::Core, alloc::alloc::Global>, core::option::Option<()>)>
2024-11-15T20:11:34.965Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:283:12
2024-11-15T20:11:34.966Z	  22: std::thread::local::LocalKey<T>::with
2024-11-15T20:11:34.966Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:260:9
2024-11-15T20:11:34.966Z	  23: tokio::runtime::context::set_scheduler
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:9
2024-11-15T20:11:34.967Z	  24: tokio::runtime::scheduler::current_thread::CoreGuard::enter
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:27
2024-11-15T20:11:34.967Z	  25: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:683:19
2024-11-15T20:11:34.967Z	  26: {closure#0}<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:191:28
2024-11-15T20:11:34.967Z	  27: tokio::runtime::context::runtime::enter_runtime
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/runtime.rs:65:16
2024-11-15T20:11:34.967Z	  28: block_on<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:179:9
2024-11-15T20:11:34.967Z	  29: tokio::runtime::runtime::Runtime::block_on_inner
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:361:47
2024-11-15T20:11:34.967Z	  30: block_on<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.967Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:335:13
2024-11-15T20:11:34.968Z	  31: test_mgs_metrics
2024-11-15T20:11:34.968Z	             at ./tests/integration_tests/metrics.rs:530:1
2024-11-15T20:11:34.968Z	  32: test_all::integration_tests::metrics::test_mgs_metrics::{{closure}}
2024-11-15T20:11:34.968Z	             at ./tests/integration_tests/metrics.rs:533:2
2024-11-15T20:11:34.968Z	  33: core::ops::function::FnOnce::call_once
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
2024-11-15T20:11:34.968Z	  34: core::ops::function::FnOnce::call_once
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
2024-11-15T20:11:34.968Z	  35: test::__rust_begin_short_backtrace
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/test/src/lib.rs:624:18
2024-11-15T20:11:34.968Z	  36: test::run_test_in_process::{{closure}}
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/test/src/lib.rs:647:60
2024-11-15T20:11:34.968Z	  37: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panic/unwind_safe.rs:272:9
2024-11-15T20:11:34.968Z	  38: std::panicking::try::do_call
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:557:40
2024-11-15T20:11:34.968Z	  39: std::panicking::try
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:521:19
2024-11-15T20:11:34.968Z	  40: std::panic::catch_unwind
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panic.rs:350:14
2024-11-15T20:11:34.968Z	  41: test::run_test_in_process
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/test/src/lib.rs:647:27
2024-11-15T20:11:34.968Z	  42: test::run_test::{{closure}}
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/test/src/lib.rs:568:43
2024-11-15T20:11:34.968Z	  43: test::run_test::{{closure}}
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/test/src/lib.rs:598:41
2024-11-15T20:11:34.968Z	  44: std::sys::backtrace::__rust_begin_short_backtrace
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/sys/backtrace.rs:152:18
2024-11-15T20:11:34.968Z	  45: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/mod.rs:538:17
2024-11-15T20:11:34.968Z	  46: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panic/unwind_safe.rs:272:9
2024-11-15T20:11:34.968Z	  47: std::panicking::try::do_call
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:557:40
2024-11-15T20:11:34.968Z	  48: std::panicking::try
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:521:19
2024-11-15T20:11:34.968Z	  49: std::panic::catch_unwind
2024-11-15T20:11:34.968Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panic.rs:350:14
2024-11-15T20:11:34.968Z	  50: std::thread::Builder::spawn_unchecked_::{{closure}}
2024-11-15T20:11:34.969Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/mod.rs:537:30
2024-11-15T20:11:34.969Z	  51: core::ops::function::FnOnce::call_once{{vtable.shim}}
2024-11-15T20:11:34.969Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
2024-11-15T20:11:34.969Z	  52: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
2024-11-15T20:11:34.972Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/alloc/src/boxed.rs:2070:9
2024-11-15T20:11:34.972Z	  53: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
2024-11-15T20:11:34.972Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/alloc/src/boxed.rs:2070:9
2024-11-15T20:11:34.972Z	  54: std::sys::pal::unix::thread::Thread::new::thread_start
2024-11-15T20:11:34.972Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/sys/pal/unix/thread.rs:108:17
2024-11-15T20:11:34.973Z	  55: <unknown>
2024-11-15T20:11:34.973Z	  56: <unknown>
2024-11-15T20:11:34.973Z	query: get hardware_component:temperature
2024-11-15T20:11:34.973Z	response: TestResponse {
2024-11-15T20:11:34.973Z	    status: 400,
2024-11-15T20:11:34.973Z	    headers: {
2024-11-15T20:11:34.973Z	        "content-type": "application/json",
2024-11-15T20:11:34.973Z	        "x-request-id": "2b2d395e-3586-4825-8384-f5935e8a8e9d",
2024-11-15T20:11:34.973Z	        "content-length": "165",
2024-11-15T20:11:34.973Z	        "date": "Fri, 15 Nov 2024 20:11:34 GMT",
2024-11-15T20:11:34.973Z	    },
2024-11-15T20:11:34.973Z	    body: b"{\n  \"request_id\": \"2b2d395e-3586-4825-8384-f5935e8a8e9d\",\n  \"error_code\": \"InvalidRequest\",\n  \"message\": \"Timeseries not found for: hardware_component:temperature\"\n}",
2024-11-15T20:11:34.973Z	}
2024-11-15T20:11:34.973Z	stack backtrace:
2024-11-15T20:11:34.973Z	   0: rust_begin_unwind
2024-11-15T20:11:34.974Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
2024-11-15T20:11:34.974Z	   1: core::panicking::panic_fmt
2024-11-15T20:11:34.974Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:74:14
2024-11-15T20:11:34.974Z	   2: {closure#1}<&&str>
2024-11-15T20:11:34.974Z	             at ./tests/integration_tests/metrics.rs:315:13
2024-11-15T20:11:34.974Z	   3: core::result::Result<T,E>::unwrap_or_else
2024-11-15T20:11:34.974Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/result.rs:1456:23
2024-11-15T20:11:34.974Z	   4: {async_fn#0}<&&str>
2024-11-15T20:11:34.974Z	             at ./tests/integration_tests/metrics.rs:313:5
2024-11-15T20:11:34.974Z	   5: {async_fn#0}
2024-11-15T20:11:34.974Z	             at ./tests/integration_tests/metrics.rs:693:18
2024-11-15T20:11:34.974Z	   6: {async_block#0}
2024-11-15T20:11:34.974Z	             at ./tests/integration_tests/metrics.rs:659:79
2024-11-15T20:11:34.974Z	   7: {async_fn#0}<(), (), test_all::integration_tests::metrics::test_mgs_metrics::{async_block#0}::test_mgs_metrics::{async_fn#0}::check_all_timeseries_present::{async_fn#0}::{closure_env#1}, test_all::integration_tests::metrics::test_mgs_metrics::{async_block#0}::test_mgs_metrics::{async_fn#0}::check_all_timeseries_present::{async_fn#0}::{closure#1}::{async_block_env#0}>
2024-11-15T20:11:34.974Z	             at /work/oxidecomputer/omicron/test-utils/src/dev/poll.rs:80:28
2024-11-15T20:11:34.975Z	   8: {async_fn#0}
2024-11-15T20:11:34.975Z	             at ./tests/integration_tests/metrics.rs:670:10
2024-11-15T20:11:34.975Z	   9: {async_fn#0}
2024-11-15T20:11:34.975Z	             at ./tests/integration_tests/metrics.rs:744:75
2024-11-15T20:11:34.975Z	  10: {async_block#0}
2024-11-15T20:11:34.975Z	             at ./tests/integration_tests/metrics.rs:530:1
2024-11-15T20:11:34.975Z	  11: poll<&mut dyn core::future::future::Future<Output=()>>
2024-11-15T20:11:34.975Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
2024-11-15T20:11:34.975Z	  12: poll<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.975Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
2024-11-15T20:11:34.975Z	  13: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:57
2024-11-15T20:11:34.975Z	  14: with_budget<core::task::poll::Poll<()>, tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure#0}::{closure#0}::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>>
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:107:5
2024-11-15T20:11:34.975Z	  15: budget<core::task::poll::Poll<()>, tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure#0}::{closure#0}::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>>
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:73:5
2024-11-15T20:11:34.975Z	  16: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:25
2024-11-15T20:11:34.975Z	  17: tokio::runtime::scheduler::current_thread::Context::enter
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:423:19
2024-11-15T20:11:34.975Z	  18: {closure#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:695:36
2024-11-15T20:11:34.975Z	  19: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:68
2024-11-15T20:11:34.975Z	  20: tokio::runtime::context::scoped::Scoped<T>::set
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/scoped.rs:40:9
2024-11-15T20:11:34.975Z	  21: tokio::runtime::context::set_scheduler::{{closure}}
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:26
2024-11-15T20:11:34.975Z	  22: try_with<tokio::runtime::context::Context, tokio::runtime::context::set_scheduler::{closure_env#0}<(alloc::boxed::Box<tokio::runtime::scheduler::current_thread::Core, alloc::alloc::Global>, core::option::Option<()>), tokio::runtime::scheduler::current_thread::{impl#8}::enter::{closure_env#0}<tokio::runtime::scheduler::current_thread::{impl#8}::block_on::{closure_env#0}<core::pin::Pin<&mut core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>>, core::option::Option<()>>>, (alloc::boxed::Box<tokio::runtime::scheduler::current_thread::Core, alloc::alloc::Global>, core::option::Option<()>)>
2024-11-15T20:11:34.975Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:283:12
2024-11-15T20:11:34.975Z	  23: std::thread::local::LocalKey<T>::with
2024-11-15T20:11:34.975Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:260:9
2024-11-15T20:11:34.975Z	  24: tokio::runtime::context::set_scheduler
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:9
2024-11-15T20:11:34.975Z	  25: tokio::runtime::scheduler::current_thread::CoreGuard::enter
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:27
2024-11-15T20:11:34.975Z	  26: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
2024-11-15T20:11:34.975Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:683:19
2024-11-15T20:11:34.975Z	  27: {closure#0}<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.976Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:191:28
2024-11-15T20:11:34.976Z	  28: tokio::runtime::context::runtime::enter_runtime
2024-11-15T20:11:34.976Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/runtime.rs:65:16
2024-11-15T20:11:34.976Z	  29: block_on<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.976Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:179:9
2024-11-15T20:11:34.976Z	  30: tokio::runtime::runtime::Runtime::block_on_inner
2024-11-15T20:11:34.976Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:361:47
2024-11-15T20:11:34.976Z	  31: block_on<core::pin::Pin<&mut dyn core::future::future::Future<Output=()>>>
2024-11-15T20:11:34.976Z	             at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:335:13
2024-11-15T20:11:34.976Z	  32: test_mgs_metrics
2024-11-15T20:11:34.977Z	             at ./tests/integration_tests/metrics.rs:530:1
2024-11-15T20:11:34.977Z	  33: test_all::integration_tests::metrics::test_mgs_metrics::{{closure}}
2024-11-15T20:11:34.977Z	             at ./tests/integration_tests/metrics.rs:533:2
2024-11-15T20:11:34.977Z	  34: core::ops::function::FnOnce::call_once
2024-11-15T20:11:34.977Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
2024-11-15T20:11:34.977Z	  35: core::ops::function::FnOnce::call_once
2024-11-15T20:11:34.977Z	             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
2024-11-15T20:11:34.977Z	note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
2024-11-15T20:11:34.977Z	WARN: dropped CockroachInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
2024-11-15T20:11:34.977Z	WARN: temporary directory leaked: "/var/tmp/omicron_tmp/.tmphYxkpE"
2024-11-15T20:11:34.977Z		If you would like to access the database for debugging, run the following:
2024-11-15T20:11:34.977Z	
2024-11-15T20:11:34.977Z		# Run the database
2024-11-15T20:11:34.977Z		cargo xtask db-dev run --no-populate --store-dir "/var/tmp/omicron_tmp/.tmphYxkpE/data"
2024-11-15T20:11:34.977Z		# Access the database. Note the port may change if you run multiple databases.
2024-11-15T20:11:34.977Z		cockroach sql --host=localhost:32221 --insecure
2024-11-15T20:11:34.980Z	WARN: dropped ClickHouse process without cleaning it up first (there may still be a child process running (PID 155515) and a temporary directory leaked, /var/tmp/omicron_tmp/test_all-2d920668872ead29-test_mgs_metrics.155440.1-clickhouse-wB5jkX)
2024-11-15T20:11:34.980Z	WARN: dropped DendriteInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
2024-11-15T20:11:34.980Z	WARN: dendrite temporary directory leaked: /var/tmp/omicron_tmp/.tmpxy6y9a
2024-11-15T20:11:34.980Z	WARN: dropped DendriteInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
2024-11-15T20:11:34.980Z	WARN: dendrite temporary directory leaked: /var/tmp/omicron_tmp/.tmpoEkEAZ
2024-11-15T20:11:34.981Z	WARN: dropped MgdInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
2024-11-15T20:11:34.981Z	WARN: mgd temporary directory leaked: /var/tmp/omicron_tmp/.tmpHtdYYm
2024-11-15T20:11:34.981Z	WARN: dropped MgdInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
2024-11-15T20:11:34.981Z	WARN: mgd temporary directory leaked: /var/tmp/omicron_tmp/.tmprEJHgA
```

**Comment by hawkw** [comment]
Hmm, this looks like another instance of the test trying to make assertions about timeseries before they've been collected? I'll take a look at the logs and try to figure out what's going on here.

**Comment by iliana** [comment]
also seen at https://github.com/oxidecomputer/omicron/runs/33395459887

**Comment by bnaecker** [comment]
Closed by #7156
